src/handlers/check_commits/no_merges.rs

```diff
@@ -0,0 +1,105 @@
+//! Purpose: When opening a PR, or pushing new changes, check for merge commits
+//! and notify the user of our no-merge policy.
+
+use crate::{
+    config::NoMergesConfig,
+    github::{GithubCommit, IssuesEvent},
+};
+use std::collections::HashSet;
+use std::fmt::Write;
+
+pub(super) fn merges_in_commits(
+    event: &IssuesEvent,
+    config: &NoMergesConfig,
+    commits: &Vec<GithubCommit>,
+) -> Option<(String, Vec<String>)> {
+    // Don't trigger if the PR has any of the excluded title segments.
+    if config
+        .exclude_titles
+        .iter()
+        .any(|s| event.issue.title.contains(s))
+    {
+        return None;
+    }
+
+    let mut merge_commits = HashSet::new();
```

Comment:
Let's uset `BTreeSet` to have a deterministic order. Having a random order could break the check for "do we have the same warning as the last time?" even if the commits don't change.

Reply:
Done and added the some end-to-end tests.